diff --git a/FutureSearch.s b/../FutureSearch.s
index baf8f0b..c6aa80a 100644
--- a/FutureSearch.s
+++ b/../FutureSearch.s
@@ -84,6 +84,11 @@ label000000000141: ; inside addInfo$1, ; referenced by 1 spots
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -229,6 +234,11 @@ label000000000455: ; inside addRanking$1, ; referenced by 1 spots
 	push op
 	li ax, "TIMELINE IMPACT RATING: "
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -251,6 +261,11 @@ label000000000455: ; inside addRanking$1, ; referenced by 1 spots
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -278,9 +293,25 @@ addDates$6: ; 6 args
 	ptrstack 8
 	memread4 ax
 	ptrinit ax
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrset ax
 	ptrstack 20
 	memread4 ax
 	ptrinit ax
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrset ax
 	sourceline 57
 	mr mar, sp
 	zeromem 4
@@ -493,6 +524,11 @@ addEmployment$3: ; 3 args
 	pop mar
 	ptrget ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 12
 	memread4 ax
 	farpush ax
@@ -567,6 +603,11 @@ addCauseOfDeath$1: ; 1 args
 	push op
 	li ax, "CAUSE OF DEATH:["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -589,6 +630,11 @@ addCauseOfDeath$1: ; 1 args
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -637,6 +683,11 @@ addSpouse$1: ; 1 args
 	push op
 	li ax, "KNOWN CLUSTERMATES:["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -659,6 +710,11 @@ addSpouse$1: ; 1 args
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -713,6 +769,11 @@ addKids$3: ; 3 args
 	push op
 	li ax, "KNOWN CHILDREN:["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -735,6 +796,11 @@ addKids$3: ; 3 args
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -786,6 +852,11 @@ addKids$3: ; 3 args
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -838,6 +909,11 @@ label000000001666: ; inside addKids$3, ; referenced by 1 spots
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -891,6 +967,11 @@ addHomes$1: ; 1 args
 	push op
 	li ax, "KNOWN RESIDENCES:["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -913,6 +994,11 @@ addHomes$1: ; 1 args
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -957,6 +1043,14 @@ addButton$3: ; 3 args
 	ptrstack 12
 	memread4 ax
 	ptrinit ax
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrset ax
 	sourceline 143
 	li ax, "   "
 	newstr ax
@@ -1039,7 +1133,7 @@ getNoteString$1: ; 1 args
 	jzi label000000002168
 	li ax, "Joseph"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002168: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 167
 	ptrstack 8
@@ -1052,7 +1146,7 @@ label000000002168: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002195
 	li ax, "Anderson"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002195: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 168
 	ptrstack 8
@@ -1065,7 +1159,7 @@ label000000002195: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002222
 	li ax, "Celia"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002222: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 169
 	ptrstack 8
@@ -1078,7 +1172,7 @@ label000000002222: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002249
 	li ax, "Rastar"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002249: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 170
 	ptrstack 8
@@ -1091,7 +1185,7 @@ label000000002249: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002276
 	li ax, "Lisa"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002276: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 171
 	ptrstack 8
@@ -1104,7 +1198,7 @@ label000000002276: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002303
 	li ax, "Sandman"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002303: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 172
 	ptrstack 8
@@ -1117,7 +1211,7 @@ label000000002303: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002330
 	li ax, "RLS-332"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002330: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 173
 	ptrstack 8
@@ -1130,7 +1224,7 @@ label000000002330: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002357
 	li ax, "Rhona"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002357: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 174
 	ptrstack 8
@@ -1143,7 +1237,7 @@ label000000002357: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002384
 	li ax, "Foster"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002384: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 175
 	ptrstack 8
@@ -1156,7 +1250,7 @@ label000000002384: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002411
 	li ax, "Oliver"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002411: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 177
 	ptrstack 8
@@ -1169,7 +1263,7 @@ label000000002411: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002438
 	li ax, "Jackson"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002438: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 178
 	ptrstack 8
@@ -1182,7 +1276,7 @@ label000000002438: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002465
 	li ax, "Grant"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002465: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 179
 	ptrstack 8
@@ -1195,7 +1289,7 @@ label000000002465: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002492
 	li ax, "Elizabeth"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002492: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 180
 	ptrstack 8
@@ -1208,7 +1302,7 @@ label000000002492: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002519
 	li ax, "Arthur"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002519: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 181
 	ptrstack 8
@@ -1221,7 +1315,7 @@ label000000002519: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002546
 	li ax, "Camron"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002546: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 182
 	ptrstack 8
@@ -1234,7 +1328,7 @@ label000000002546: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002573
 	li ax, "Olivia"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002573: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 183
 	ptrstack 8
@@ -1247,7 +1341,7 @@ label000000002573: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002600
 	li ax, "Elizabeth"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002600: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 184
 	ptrstack 8
@@ -1260,7 +1354,7 @@ label000000002600: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002627
 	li ax, "NYPD"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002627: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 185
 	ptrstack 8
@@ -1273,7 +1367,7 @@ label000000002627: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002654
 	li ax, "6-304"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002654: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 186
 	ptrstack 8
@@ -1286,7 +1380,7 @@ label000000002654: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002681
 	li ax, "Mitchell"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002681: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 187
 	ptrstack 8
@@ -1299,7 +1393,7 @@ label000000002681: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002708
 	li ax, "Santina"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002708: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 188
 	ptrstack 8
@@ -1312,7 +1406,7 @@ label000000002708: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002735
 	li ax, "Murial"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002735: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 189
 	ptrstack 8
@@ -1325,7 +1419,7 @@ label000000002735: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002762
 	li ax, "Pie Sycamore"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002762: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 191
 	ptrstack 8
@@ -1338,7 +1432,7 @@ label000000002762: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002789
 	li ax, "Eddie"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002789: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 192
 	ptrstack 8
@@ -1351,7 +1445,7 @@ label000000002789: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002816
 	li ax, "Ida"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002816: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 193
 	ptrstack 8
@@ -1364,7 +1458,7 @@ label000000002816: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002843
 	li ax, "Paula"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002843: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 194
 	ptrstack 8
@@ -1377,7 +1471,7 @@ label000000002843: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002870
 	li ax, "Patrick"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002870: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 195
 	ptrstack 8
@@ -1390,7 +1484,7 @@ label000000002870: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002897
 	li ax, "Brian"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002897: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 196
 	ptrstack 8
@@ -1403,7 +1497,7 @@ label000000002897: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002924
 	li ax, "Inman"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002924: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 197
 	ptrstack 8
@@ -1416,7 +1510,7 @@ label000000002924: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002951
 	li ax, "Simon"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002951: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 198
 	ptrstack 8
@@ -1429,7 +1523,7 @@ label000000002951: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000002978
 	li ax, "Clancy"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000002978: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 199
 	ptrstack 8
@@ -1442,7 +1536,7 @@ label000000002978: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003005
 	li ax, "Sieveri"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003005: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 200
 	ptrstack 8
@@ -1455,7 +1549,7 @@ label000000003005: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003032
 	li ax, "Breakdown Boys"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003032: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 201
 	ptrstack 8
@@ -1468,7 +1562,7 @@ label000000003032: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003059
 	li ax, "Brian"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003059: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 202
 	ptrstack 8
@@ -1481,7 +1575,7 @@ label000000003059: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003086
 	li ax, "Mason"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003086: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 203
 	ptrstack 8
@@ -1494,7 +1588,7 @@ label000000003086: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003113
 	li ax, "Chloe"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003113: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 204
 	ptrstack 8
@@ -1507,7 +1601,7 @@ label000000003113: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003140
 	li ax, "Sadar"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003140: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 206
 	ptrstack 8
@@ -1520,7 +1614,7 @@ label000000003140: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003167
 	li ax, "Sam"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003167: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 207
 	ptrstack 8
@@ -1533,7 +1627,7 @@ label000000003167: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003194
 	li ax, "Deane"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003194: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 208
 	ptrstack 8
@@ -1546,7 +1640,7 @@ label000000003194: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003221
 	li ax, "Sterling Agency"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003221: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 210
 	ptrstack 8
@@ -1559,7 +1653,7 @@ label000000003221: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003248
 	li ax, "Winera"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003248: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 211
 	ptrstack 8
@@ -1572,7 +1666,7 @@ label000000003248: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003275
 	li ax, "Hanna"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003275: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 212
 	ptrstack 8
@@ -1585,7 +1679,7 @@ label000000003275: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003302
 	li ax, "Imani"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003302: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 213
 	ptrstack 8
@@ -1607,7 +1701,7 @@ label000000003302: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003350
 	li ax, "Nash"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003350: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 216
 	ptrstack 8
@@ -1620,7 +1714,7 @@ label000000003350: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003377
 	li ax, "Finley"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003377: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 217
 	ptrstack 8
@@ -1633,7 +1727,7 @@ label000000003377: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003404
 	li ax, "Quentin"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003404: ; inside getNoteString$1, ; referenced by 2 spots
 	sourceline 219
 	ptrstack 8
@@ -1646,7 +1740,7 @@ label000000003404: ; inside getNoteString$1, ; referenced by 2 spots
 	jzi label000000003431
 	li ax, "Yvonne"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003431: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 220
 	ptrstack 8
@@ -1659,7 +1753,7 @@ label000000003431: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003458
 	li ax, "Dion"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003458: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 221
 	ptrstack 8
@@ -1672,7 +1766,7 @@ label000000003458: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003485
 	li ax, "Lozzotti"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003485: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 222
 	ptrstack 8
@@ -1685,7 +1779,7 @@ label000000003485: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003512
 	li ax, "Winera"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003512: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 223
 	ptrstack 8
@@ -1698,7 +1792,7 @@ label000000003512: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003539
 	li ax, "Tanaka"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003539: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 224
 	ptrstack 8
@@ -1711,7 +1805,7 @@ label000000003539: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003566
 	li ax, "Hanna"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003566: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 225
 	ptrstack 8
@@ -1724,7 +1818,7 @@ label000000003566: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003593
 	li ax, "Nicholas"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003593: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 226
 	ptrstack 8
@@ -1737,7 +1831,7 @@ label000000003593: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003620
 	li ax, "Chord"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003620: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 228
 	ptrstack 8
@@ -1750,7 +1844,7 @@ label000000003620: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003647
 	li ax, "Adam"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003647: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 229
 	ptrstack 8
@@ -1763,7 +1857,7 @@ label000000003647: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003674
 	li ax, "Cole"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003674: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 230
 	ptrstack 8
@@ -1776,7 +1870,7 @@ label000000003674: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003701
 	li ax, "Otis"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003701: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 231
 	ptrstack 8
@@ -1789,7 +1883,7 @@ label000000003701: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003728
 	li ax, "Knox"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003728: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 232
 	ptrstack 8
@@ -1802,7 +1896,7 @@ label000000003728: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003755
 	li ax, "Marla"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003755: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 233
 	ptrstack 8
@@ -1815,7 +1909,7 @@ label000000003755: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003782
 	li ax, "Liam"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003782: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 234
 	ptrstack 8
@@ -1828,7 +1922,7 @@ label000000003782: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003809
 	li ax, "Whiteman"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003809: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 235
 	ptrstack 8
@@ -1841,7 +1935,7 @@ label000000003809: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003836
 	li ax, "Hotel Commonwealth"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003836: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 236
 	ptrstack 8
@@ -1854,7 +1948,7 @@ label000000003836: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003863
 	li ax, "Quanmexlin"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003863: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 237
 	ptrstack 8
@@ -1867,7 +1961,7 @@ label000000003863: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003890
 	li ax, "Evelyn"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003890: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 238
 	ptrstack 8
@@ -1880,7 +1974,7 @@ label000000003890: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003917
 	li ax, "Sherman"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003917: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 240
 	ptrstack 8
@@ -1893,7 +1987,7 @@ label000000003917: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003944
 	li ax, "Field Manual"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003944: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 241
 	ptrstack 8
@@ -1906,7 +2000,7 @@ label000000003944: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003971
 	li ax, "Nash"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003971: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 242
 	ptrstack 8
@@ -1919,7 +2013,7 @@ label000000003971: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000003998
 	li ax, "Quentin"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000003998: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 243
 	ptrstack 8
@@ -1932,11 +2026,21 @@ label000000003998: ; inside getNoteString$1, ; referenced by 1 spots
 	jzi label000000004025
 	li ax, "Finley"
 	newstr ax
-	ret 
+	jmpi label_translate_note
 label000000004025: ; inside getNoteString$1, ; referenced by 1 spots
 	sourceline 244
 	li ax, 0
 	ret 
+;# call GetTranslation on note string before return
+label_translate_note:
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ret
 
 doFlash$1: ; 1 args
 	sourceline 250
@@ -2838,6 +2942,11 @@ raGetMitchellRecord$0: ; 0 args
 	jzi label000000006053
 	sourceline 370
 	li ax, "Disciplinary record:[Exemplary"
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	ptrstack 4
 	push mar
@@ -2848,6 +2957,11 @@ raGetMitchellRecord$0: ; 0 args
 label000000006053: ; inside raGetMitchellRecord$0, ; referenced by 1 spots
 	sourceline 374
 	li ax, "Disciplinary record: 1 count of negligence (Patrol car accident)[[Details: Officer\'s patrol car collided with a fire hydrant at 63rd and Lexington. No one was injured at the scene. Officer judged to be at fault."
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	ptrstack 4
 	push mar
@@ -2876,6 +2990,11 @@ label000000006053: ; inside raGetMitchellRecord$0, ; referenced by 1 spots
 label000000006115: ; inside raGetMitchellRecord$0, ; referenced by 2 spots
 	sourceline 377
 	li ax, "Mitchell Santina[ID: 6-304[[Districts served:[Midtown South: 1908 - 1931[Staten Island: 1931 - 1942[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	mr mar, sp
 	ptrinit ax
@@ -2888,6 +3007,11 @@ label000000006115: ; inside raGetMitchellRecord$0, ; referenced by 2 spots
 	ptrget ax
 	assert ax
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -4710,6 +4834,14 @@ label000000010261: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "field manual"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4734,6 +4866,14 @@ label000000010348: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "rastar"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4754,6 +4894,14 @@ label000000010392: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "joseph anderson"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4765,6 +4913,14 @@ label000000010392: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "anderson joseph"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4812,6 +4968,14 @@ label000000010524: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "celia anderson"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4823,6 +4987,14 @@ label000000010524: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "anderson celia"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4899,6 +5071,14 @@ label000000010722: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "lisa sandman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -4910,6 +5090,14 @@ label000000010722: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "sandman lisa"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5008,6 +5196,14 @@ label000000010978: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "rhona foster"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5019,6 +5215,14 @@ label000000010978: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "foster rhona"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5094,6 +5298,14 @@ label000000011179: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "oliver foster"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5105,6 +5317,14 @@ label000000011179: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "foster oliver"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5211,6 +5431,14 @@ label000000011444: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "patrick inman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5222,6 +5450,14 @@ label000000011444: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "inman patrick"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5246,6 +5482,14 @@ label000000011521: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "eddie inman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5289,6 +5533,14 @@ label000000011618: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "paula inman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5300,6 +5552,14 @@ label000000011618: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "inman paula"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5386,6 +5646,14 @@ label000000011837: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "simon inman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5397,6 +5665,14 @@ label000000011837: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "inman simon"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5467,6 +5743,14 @@ label000000012019: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "jackson grant"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5478,6 +5762,14 @@ label000000012019: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "grant jackson"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5614,6 +5906,14 @@ label000000012365: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "olivia grant"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5625,6 +5925,14 @@ label000000012365: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "grant olivia"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5662,6 +5970,11 @@ label000000012421: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "EMPLOYMENT HISTORY:[Unknown[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -5733,6 +6046,14 @@ label000000012642: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "elizabeth grant"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5744,6 +6065,14 @@ label000000012642: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "grant elizabeth"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5830,6 +6159,14 @@ label000000012861: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "elizabeth camron"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5841,6 +6178,14 @@ label000000012861: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "camron elizabeth"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5963,6 +6308,14 @@ label000000013167: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "arthur camron"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -5974,6 +6327,14 @@ label000000013167: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "camron arthur"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6096,6 +6457,14 @@ label000000013473: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "nypd"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6311,6 +6680,14 @@ label000000013956: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "murial santina"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6322,6 +6699,14 @@ label000000013956: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "santina murial"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6377,6 +6762,11 @@ label000000014012: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "Bibliography:[\"The Curse of Captain Billop\", July 1953[\"Hunting Richmond County\", March 1958[\"Last Beer in Oakwood\", Nov 1963[\"The Fool of Prince\'s Bay\", Feb 1968[\"Winning in Willowbrook\", October 1970[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -6429,6 +6819,14 @@ label000000014230: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "pie sycamore"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6582,6 +6980,14 @@ label000000014585: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "chloe sadar"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6593,6 +6999,14 @@ label000000014585: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "sadar chloe"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6844,6 +7258,14 @@ label000000015189: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "mitchell santina"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6855,6 +7277,14 @@ label000000015189: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "santina mitchell"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6879,6 +7309,14 @@ label000000015266: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "clancy sieveri"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6923,6 +7361,14 @@ label000000015365: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "breakdown boys"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -6936,6 +7382,11 @@ label000000015365: ; inside futureSearch$1, ; referenced by 1 spots
 	subi sp, 4
 	sourceline 959
 	li ax, "Breakdown Boys Gang[Dates of operation: 1852 - 1875 (estimated)[[Led by the mobster known as Eddie the Bear, the Breakdown Boys rose to become one of New York\'s most feared criminal organizations. As the kings of Manhattan\'s seaports, the \"boys\" raked in huge profits running brothels, protection rackets, drug rings and even murder-for-hire operations.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -6945,6 +7396,11 @@ label000000015365: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "The identity of Eddie the Bear is still unknown, but it is estimated he died in the mid 1870s. Infighting over who should replace him led to the gang\'s power and influence weakening until the gang disappeared from the historical record by 1875.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -6963,6 +7419,11 @@ label000000015365: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "Historical evidence suggests that the nerve-center of the gang\'s operation was the Badger\'s Fish Market at the edge of South Street Seaport, now a popular tourist spot.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -7040,6 +7501,14 @@ label000000015631: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "brian mason"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7051,6 +7520,14 @@ label000000015631: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "mason brian"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7161,6 +7638,14 @@ label000000015915: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "sam deane"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7172,6 +7657,14 @@ label000000015915: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "deane sam"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7216,6 +7709,11 @@ label000000015971: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "Born:[Feb 17, 2014 -[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -7234,6 +7732,11 @@ label000000015971: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "EMPLOYMENT HISTORY:[Office temp, Leaves (2042 - 2042)[Janitor, Burger Heaven (2053 - present)[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -7266,6 +7769,11 @@ label000000015971: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "ARREST RECORD:[[1 count of corporate espionage[Plaintiff: Leaves, inc.[Time served: 10 years, 2042-2052[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -7363,6 +7871,14 @@ label000000016379: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "sterling agency"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7414,6 +7930,14 @@ label000000016495: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "yvonne lozzotti"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7425,6 +7949,14 @@ label000000016495: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "lozzotti yvonne"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7673,6 +8205,14 @@ label000000017104: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "dion lozzotti"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7684,6 +8224,14 @@ label000000017104: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "lozzotti dion"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7796,6 +8344,11 @@ label000000017359: ; inside futureSearch$1, ; referenced by 1 spots
 	subi sp, 4
 	sourceline 1078
 	li ax, "OBITUARY[Dion Lozzotti[Feb 21, 1977 - September 5th, 2059[[Dion Lozzotti, a luminary artist, has passed away on September 5th, 2059 at the age of 82 after a lengthy battle with Skimein Syndrome.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -7805,6 +8358,11 @@ label000000017359: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "Dion was best known for his series of paintings depicting smoke emanating from Ground Zero as seen from landmarks throughout post-9/11 New York. This series is often cited as an enduring testament to the city\'s resilience.[[Dion often expressed a deep affection for these landmarks long before 9/11. He would visit a single location every day for weeks or months, drawing countless sketches until he was satisfied. This, critics have said, is what imbued his work with such a unique intimacy.[[It is often speculated that the loss of his wife, Yvonne, on that fateful September day in 2001 is what ignited his artistic odyssey.[[Dion\'s canvases, vivid with emotion, graced global galleries including the Hyde Park Art Center in Chicago and the Park Gallery in New York.[[May his art and spirit forever inspire.[[In lieu of flowers, please make a donation to the Yvonne Lozzotti Foundation."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -7852,6 +8410,14 @@ label000000017519: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "nicholas chord"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -7863,6 +8429,14 @@ label000000017519: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "chord nicholas"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8015,6 +8589,14 @@ label000000017900: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "hanna tanaka"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8026,6 +8608,14 @@ label000000017900: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "tanaka hanna"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8232,6 +8822,14 @@ label000000018402: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "winera"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8273,6 +8871,14 @@ label000000018495: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "adam cole"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8399,6 +9005,14 @@ label000000018793: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "liam whiteman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8412,6 +9026,11 @@ label000000018793: ; inside futureSearch$1, ; referenced by 1 spots
 	subi sp, 4
 	sourceline 1150
 	li ax, "OBITUARY[Liam Whiteman[April 29, 1945 - June 5, 2031[[Liam Whiteman, a cherished figure in the Coney Island community and the longtime owner of a modest bookstore on the boardwalk, passed away on June 5, 2031, at the age of 86 due to complications from a respiratory infection. Known for his encyclopedic knowledge of literature and his warm, engaging conversations, Liam built a reputation as a kind and welcoming presence to all who visited his shop.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -8421,6 +9040,11 @@ label000000018793: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "Despite the rise of e-readers and the decline of physical bookstores, Liam stubbornly refused to close or sell his store, believing in the power of printed books and their ability to create connections. Over the years, he turned down several lucrative offers to buy the shop, determined to preserve it as a haven for book lovers.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -8439,6 +9063,11 @@ label000000018793: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "Liam is survived by his sister, Marla, and his nephew, Nathan. He will be deeply missed."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -8472,6 +9101,14 @@ label000000018956: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "marla knox"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8483,6 +9120,14 @@ label000000018956: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "marla cole"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8498,6 +9143,14 @@ label000000019012: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "marla whiteman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8513,6 +9166,14 @@ label000000019045: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "marla knox"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8531,6 +9192,14 @@ label000000019087: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "marla Cole"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8549,6 +9218,14 @@ label000000019127: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "marla Whiteman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8586,6 +9263,14 @@ label000000019208: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "evelyn sherman"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8612,6 +9297,11 @@ label000000019208: ; inside futureSearch$1, ; referenced by 1 spots
 	subi sp, 4
 	sourceline 1168
 	li ax, "OBITUARY[[Evelyn Sherman passed away in the village of Jibhi on March 3, 2022, at the age of 70.[[Evelyn\'s early life was defined by sports of all kinds, and by the age of 22 she had become one of the era\'s most renowned stuntwomen. Her daring feats in films span the high-speed car chases of 1973\'s \"Steel Siren\" and the iconic rooftop jump of 1976\'s \"Valkyrie Justice\". She was known for being a perfectionist as well as fearless, often demanding (and receiving) reshoots of stunts she felt were anything less than her best.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -8621,6 +9311,11 @@ label000000019208: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "By the mid-1980s, injuries accumulated during her stunt career forced her to retire. Most knew her as a quiet soul, often tending to her beloved garden and staying away from the spotlight; but Evelyn\'s appetite for adventure never waned. She turned her attention to exploring uncharted jungles and forests, discovering rare plant and animal species. Her many discoveries include the glintfrog, the amerwing prowler, and the peruvian fernstrider.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -8639,6 +9334,11 @@ label000000019208: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "On an expedition to the Himalayas, she selflessly saved two of her fellow travelers from a dangerous fall, carrying them several dozen miles to safety. Tragically, she contracted pneumonia and died a week later.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -8657,6 +9357,11 @@ label000000019208: ; inside futureSearch$1, ; referenced by 1 spots
 	push op
 	li ax, "She finished her life as she lived it: fearless, demanding perfection, and living out her passion for the unknown. She will be greatly missed, but not soon forgotten."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -8712,6 +9417,14 @@ label000000019491: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "hotel commonwealth"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8776,6 +9489,14 @@ label000000019640: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "otis knox"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -8796,6 +9517,14 @@ label000000019684: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "quanmexlin"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -9013,6 +9742,14 @@ label000000020195: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "imani nash"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -9024,6 +9761,14 @@ label000000020195: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "imani quentin"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -9039,6 +9784,14 @@ label000000020251: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "imani finley"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -9054,6 +9807,14 @@ label000000020284: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrget ax
 	push ax
 	li ax, "imani camron"
+
+;# call GetTranslation for search entry comparison
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	pop bx
 	streq bx, ax
 	mr ax, bx
@@ -9405,6 +10166,15 @@ label000000021072: ; inside futureSearch$1, ; referenced by 1 spots
 	ptrset ax
 label000000021107: ; inside futureSearch$1, ; referenced by 1 spots
 	sourceline 1265
+	ptrstack 4
+	ptrget ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrset ax
 	li ax, 11
 	mr bx, ax
 	li mar, lbl_search_results
@@ -10430,6 +11200,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "Welcome to ChronoZen, where time travel is not just a service but a responsibility.[[In this manual, we\'ll guide you through the rules, regulations, and intricacies of navigating the temporal landscape. Please read carefully to ensure a smooth journey through time."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10448,6 +11223,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "[[1. About ChronoZen"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10466,6 +11246,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "[------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10484,6 +11269,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "In the current year (2064), time travel is a reality governed by ChronoZen, a corporation established to regulate and control temporal excursions. Due to the potential for abuse, the government entrusted ChronoZen with the authority to oversee all time travel operations."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10502,6 +11292,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "[[2. Booking a Trip"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10520,6 +11315,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "[-----------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10538,6 +11338,11 @@ fieldGuideSet$1: ; 1 args
 	push op
 	li ax, "Private citizens can embark on time travel journeys, accompanied by trained agents to ensure responsible temporal exploration. Both agents and clients are equipped with a Paradox Field Excluder (PFE), safeguarding them from the effects of time paradoxes."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10568,6 +11373,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "3. The ChronoZen Team"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10586,6 +11396,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[---------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10604,6 +11419,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "Each time travel expedition is monitored by a dedicated agent and overseen by a Monitor in the present. The Board of Directors, internally known as \"The Brass,\" formulates policies, enforces rules, and handles inquiries. The Brass ensures that each client\'s mission is carefully evaluated to minimize potential damage to the timeline."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10622,6 +11442,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[[4. Paradox Field Excluder (PFE)"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10640,6 +11465,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[-------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10658,6 +11488,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "The PFE is a crucial piece of equipment for time travelers. It shields them from temporal paradoxes, allowing changes to their personal history without erasing their existence. ChronoZen takes every precaution to prevent damage to the client\'s timeline, and in the event of a mishap, compensation is provided."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10676,6 +11511,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[[5. Agent Selection and Training"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10694,6 +11534,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[-------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10712,6 +11557,11 @@ label000000023650: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "Becoming a ChronoZen agent is no easy feat. Candidates must possess a specific psychological profile, capable of handling the constant rewriting of personal history. If you are reading this, congratulations on being one of the chosen few!"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10742,6 +11592,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "6. Landing Points"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10760,6 +11615,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[-----------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10778,6 +11638,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "Time travel destinations are carefully calculated to meet specific criteria. The landing point must be outdoors, open to the sky, and devoid of any observers or recording devices. Failure to meet these criteria will result in the inability to initiate time travel."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10796,6 +11661,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[[7. Time Impact Rankings"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10814,6 +11684,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[-----------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10832,6 +11707,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "To keep the timeline managable and organized; every person, event and location in history has been assigned a Timeline Impact Ranking (TIR). Please consult the TIR section of the field guide for more detailed information about how the rankings govern your interactions while traveling through time."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10850,6 +11730,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[[8. Emergency Rewind Protocol (ERP)"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10868,6 +11753,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[----------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10886,6 +11776,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "All Monitors have access to the ERP system, which allows them to rewind a field agent\'s personal timeline by up to 30 minutes. To prevent accidental abuse, the ERP system will only become available in the event of an agent\'s death."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10904,6 +11799,11 @@ label000000024017: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[[WARNING: Rewinds should only be performed on trained agents with long-term PFE exposure. Do NOT attempt on clients."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10934,6 +11834,11 @@ label000000024422: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[[Conclusion"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10952,6 +11857,11 @@ label000000024422: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "[----------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -10970,6 +11880,11 @@ label000000024422: ; inside fieldGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "As a ChronoZen time traveler, you are part of a carefully regulated system designed to balance exploration with responsibility. Always adhere to guidelines, respect the timelines, and enjoy your journey through the corridors of time responsibly.[[Safe travels!"
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11035,6 +11950,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "A Brief History of Time Travel[[NOTE: This guide has been chrono-locked to preserve our history. Should your knowledge of these events not match this chronology, contact your nearest ChronoZen branch immediately.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11053,6 +11973,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "1. Discovery of the Quansonic Particle (2041)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11071,6 +11996,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "---------------------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11089,6 +12019,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "In the year 2041, scientists made a monumental breakthrough with the discovery of the Quansonic particle, a mysterious entity that seemed to defy the laws of physics. Initial research suggested that these particles existed outside of traditional spacetime, opening the door to realms previously thought unreachable.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11107,6 +12042,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "2. The Timestream Revelation (2043)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11125,6 +12065,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "-----------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11143,6 +12088,11 @@ historyGuideSet$1: ; 1 args
 	push op
 	li ax, "Further investigation into the Quansonic particles revealed a startling truth: they were leaking from another dimension altogether, a dimension that became known as \"the timestream.\" This revelation sparked a scientific revolution as researchers embarked on a quest to understand and harness the enigmatic properties of this newfound dimension.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11173,6 +12123,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "3. The Quest for Time Travel (2044-2045)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11191,6 +12146,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "----------------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11209,6 +12169,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "With the knowledge of the timestream at their fingertips, scientists worldwide focused their efforts on devising methods to traverse this extradimensional realm. The pursuit of time travel became the holy grail of scientific exploration, with researchers racing against time to unlock the secrets of temporal manipulation.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11227,6 +12192,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "4. Breakthrough in Time Travel Technology (2045)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11245,6 +12215,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "------------------------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11263,6 +12238,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "In a historic moment for humanity, scientists succeeded in developing the first functional time-travel technology. By harnessing the power of the timestream, they created a gateway to the past, allowing for unprecedented journeys through the annals of history.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11281,6 +12261,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "5. Government Intervention and Regulations (2047)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11299,6 +12284,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "-------------------------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11317,6 +12307,11 @@ label000000024926: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "As the implications of time travel became apparent, governments intervened to regulate its usage and prevent potential abuse. Stricter regulations were implemented to ensure the responsible exploration of the timestream, with oversight from global authorities."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11347,6 +12342,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "6. Rise of Irresponsible Use (2048-2051)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11365,6 +12365,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "----------------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11383,6 +12388,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "Despite regulatory efforts, the allure of time travel proved too tempting for some. Wealthy individuals exploited corrupt scientists to satisfy personal desires, leading to chaos and damage across the temporal landscape.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11401,6 +12411,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "7. Formation of ChronoZen (2052)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11419,6 +12434,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "--------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11437,6 +12457,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "In response to the escalating crisis, governments collaborated to establish ChronoZen, a specialized corporation tasked with regulating and overseeing all time travel operations. ChronoZen\'s formation marked a turning point in the chaotic era of temporal exploration, bringing order and structure to the realm of time travel.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11455,6 +12480,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "8. Restoring Order and Responsibility (2053-Present)["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11473,6 +12503,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "----------------------------------------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11491,6 +12526,11 @@ label000000025293: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "Under the guidance of ChronoZen, humanity embarked on a journey of responsible temporal exploration. With strict regulations in place and trained agents ensuring ethical conduct, the dangers of irresponsible time travel were mitigated, allowing for the preservation of history and the protection of the temporal fabric.[["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11521,6 +12561,11 @@ label000000025660: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "9. Into the Future["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11539,6 +12584,11 @@ label000000025660: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "------------------["
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11557,6 +12607,11 @@ label000000025660: ; inside historyGuideSet$1, ; referenced by 1 spots
 	push op
 	li ax, "The discovery of the Quansonic particle and the subsequent revelation of the timestream reshaped the course of human history, opening doors to realms previously thought unreachable. As humanity continues to navigate the complexities of time travel, the lessons learned from the past serve as a beacon guiding us towards a future of responsible exploration and discovery."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	ptrstack 8
 	push mar
 	pop mar
@@ -11782,6 +12837,11 @@ label000000026187: ; inside processNozzoButton$1, ; referenced by 1 spots
 	pop op
 	sourceline 1551
 	li ax, "Rastar Locks were the most popular brand of combination padlock from 1992-2043. Rastar Lock combination padlocks were typically found securing lockers, bicycles, gates and many other security applications.[[The manufacturing of these locks was phased out in 2042, when DNA-encoded clamps became commonplace.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -12035,6 +13095,11 @@ label000000026817: ; inside processNozzoButton$1, ; referenced by 1 spots
 	jzi label000000026929
 	sourceline 1596
 	li ax, "OBITUARY[Elizabeth Camron[Feb 21, 1903 - November 12th, 1980[[Elizabeth Grant Camron, 77, died peacefully in her sleep on November 12th.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -12044,6 +13109,11 @@ label000000026817: ; inside processNozzoButton$1, ; referenced by 1 spots
 	push op
 	li ax, "Mrs. Camron worked for 36 years as a teacher and educator of middle school children, before retiring in 1962.[[In 1923 she married Arthur Camron, and they were together until Arthur\'s death in 1979.[[Elizabeth loved good food, philanthropy and socializing. She was well-known for having a willing ear, a keen mind, and a wide-open heart.[[She is survived by her sons Jack and Sam; six grandchildren; and two great-grandchildren."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -12130,6 +13200,11 @@ label000000027034: ; inside processNozzoButton$1, ; referenced by 1 spots
 	jzi label000000027136
 	sourceline 1612
 	li ax, "OBITUARY[Arthur Camron[April 14, 1900 - July 20th, 1979[[Arthur Edward Camron, a longtime resident of NY, died peacefully on July 20th at the age of 79. Art cherished his family and many friends and he always greeted strangers with warmth and kind words.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -12139,6 +13214,11 @@ label000000027034: ; inside processNozzoButton$1, ; referenced by 1 spots
 	push op
 	li ax, "Camron worked over 50 years in the urban development industry, retiring in 1971 from Seagram Real Estate to dedicate the rest of his life to philanthropy. He was an active member of the Children\'s Aid Society, and relished family time with his wife, children, and grandchildren as he travelled the world.[[Art is survived by his wife Elizabeth; his sons Jack and Sam; six grandchildren; and two great-grandchildren."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
@@ -12335,6 +13415,11 @@ label000000027513: ; inside processNozzoButton$1, ; referenced by 1 spots
 	jzi label000000027673
 	sourceline 1641
 	li ax, "OBITUARY[Mitchell Santina[March 14, 1890 - Nov 26th, 1961[[Mitchell Santina, a lifetime resident of NY, died peacefully at his Staten Island home on Nov 26th at the age of 71. Mitch was a lifetime member of the NYPD, serving as an officer from the age of 18 until he retired at the age of 52.[["
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
 	newstr ax
 	li mar, @var000000
 	push mar
@@ -12344,6 +13429,11 @@ label000000027513: ; inside processNozzoButton$1, ; referenced by 1 spots
 	push op
 	li ax, "Since his retirement, he dedicated his life to training the next generation of officers at the Richmond Terrace Police Academy.[[Mitchell is survived by his wife Murial; his three sons (all NYPD officers); three grandchildren; and a lifetime of dedicated service."
 	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	farpush ax
 	li mar, @var000000
 	push mar
 	pop mar
