diff --git a/Personic.s b/../Personic.s
index 69a3e3e..46130f4 100644
--- a/Personic.s
+++ b/../Personic.s
@@ -191,6 +191,15 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+;# call GetTranslation for argument into String::Format
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "Employed at: %s"
 	farpush ax
@@ -220,6 +229,15 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+;# call GetTranslation for argument into String::Format
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "Location: %s"
 	farpush ax
@@ -250,6 +268,15 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	add mar, cx
 	ptrget ax
 	farpush ax
+
+;# call GetTranslation for argument into String::Format
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	li ax, "Cluster status: %s"
 	farpush ax
 	setfuncargs 2
@@ -278,6 +305,15 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+;# call GetTranslation for argument into String::Format
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "About me: %s"
 	farpush ax
