diff --git a/Personic.s b/../modified/Personic.s
index 69a3e3e..b2f694a 100644
--- a/Personic.s
+++ b/../modified/Personic.s
@@ -191,6 +191,14 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "Employed at: %s"
 	farpush ax
@@ -220,6 +228,14 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "Location: %s"
 	farpush ax
@@ -249,6 +265,14 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "Cluster status: %s"
 	farpush ax
@@ -278,6 +302,14 @@ label000000000375: ; inside personicDisplay$1, ; referenced by 1 spots
 	pop mar
 	add mar, cx
 	ptrget ax
+
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+
 	farpush ax
 	li ax, "About me: %s"
 	farpush ax
