diff --git a/TalkyManager.s b/../TalkyManager.s
index de85db4..4155897 100644
--- a/TalkyManager.s
+++ b/../TalkyManager.s
@@ -26696,6 +26696,20 @@ Character::GSay$5: ; 5 args
 	mr mar, sp
 	memcpy 4, 0
 	addi sp, 4
+
+;# call GetTranslation for text argument
+	ptrstack 12
+	ptrget ax
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrstack 12
+	ptrset ax
+
 	sourceline 3294
 	li ax, pauseBGspeech$0
 	call ax
