diff --git a/typewriter.s b/../typewriter.s
index c1492d8..7c6a559 100644
--- a/typewriter.s
+++ b/../typewriter.s
@@ -14,6 +14,20 @@ Typewriter::Type$11: ; 11 args
 	ptrstack 48
 	memread4 ax
 	ptrinit ax
+
+;# call GetTranslation for text argument
+	ptrstack 28
+	ptrget ax
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrstack 28
+	ptrset ax
+
 	sourceline 8
 	li ax, ""
 	newstr ax
@@ -119,6 +133,16 @@ label000000000157: ; inside Typewriter::Type$11, ; referenced by 1 spots
 	farcall ax
 	farsubsp 1
 	pop op
+; fix transparency bug (new location)
+	li ax, 0
+	push op
+	callobj mar
+	farpush ax
+	setfuncargs 1
+	li ax, GUI::set_Transparency
+	farcall ax
+	farsubsp 1
+	pop op
 	sourceline 18
 	ptrstack 52
 	push mar
@@ -2053,6 +2077,20 @@ mySlate$1: ; 1 args
 	ptrstack 8
 	memread4 ax
 	ptrinit ax
+
+;# call GetTranslation for text argument
+	ptrstack 8
+	ptrget ax
+	assert ax
+	farpush ax
+	setfuncargs 1
+	li ax, GetTranslation
+	farcall ax
+	farsubsp 1
+	newstr ax
+	ptrstack 8
+	ptrset ax
+
 	sourceline 172
 	li ax, 1
 	farpush ax
